include/superior_mysqlpp/row_stream.hpp

```diff
@@ -0,0 +1,43 @@
+#pragma once
+#include <superior_mysqlpp/row.hpp>
+
+
+namespace SuperiorMySqlpp
+{
+    /**
+     * Adapter for Row that provides operator>>() for data extraction.
+     * Example:
+     *
+     *  auto row = ...
+     *  std::string s;
+     *  int i = 0;
+     *  SuperiorMySqlpp::RowStream {row}
+     *      >> s
+     *      >> i
+     *      ;
+     */
+    class RowStream
+    {
+    private:
+        Row::Iterator first;
+        Row::Iterator last;
+
+    public:
+        explicit RowStream(Row& row)
+            : first {row.begin()}
+            , last {row.end()}
+        {
+        }
+
+        template<typename T>
+        RowStream& operator>>(T& obj)
+        {
+            assert (first != last);
```

Comment:
This adapter is intended for callers who are aware of how many columns there are in resulting row. I can provide `operator bool()` to check if the extraction is valid for callers who do no want to keep track of the number themselves. Assert just documents this precondition.

Exception is costly. Assert is noop when NDEBUG is #defined.